File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/UserClaimMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class UserClaimMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "userClaimIdUserId",
            "userClaimIdClaimId",
            "userVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "Integer",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private UserClaimMvoStateEventId value;

    public UserClaimMvoStateEventIdFlattenedDto()
    {
        this(new UserClaimMvoStateEventId());
    }

    public UserClaimMvoStateEventIdFlattenedDto(UserClaimMvoStateEventId value)
    {
        this.value = value;
    }

    public UserClaimMvoStateEventId toUserClaimMvoStateEventId()
    {
        return this.value;
    }


    public String getUserClaimIdUserId()
    {
        return this.value.getUserClaimId().getUserId();
    }

    public void setUserClaimIdUserId(String userClaimIdUserId)
    {
        this.value.getUserClaimId().setUserId(userClaimIdUserId);
    }

    public Integer getUserClaimIdClaimId()
    {
        return this.value.getUserClaimId().getClaimId();
    }

    public void setUserClaimIdClaimId(Integer userClaimIdClaimId)
    {
        this.value.getUserClaimId().setClaimId(userClaimIdClaimId);
    }

    public Long getUserVersion()
    {
        return this.value.getUserVersion();
    }

    public void setUserVersion(Long userVersion)
    {
        this.value.setUserVersion(userVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        UserClaimMvoStateEventIdFlattenedDto other = (UserClaimMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/UserClaimMvoStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class UserClaimMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'userClaimIdUserId' => 'string',
            'userClaimIdClaimId' => 'integer',
            'userVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var UserClaimMvoStateEventId
     */
    private $value;

    /**
     * @param UserClaimMvoStateEventId $value
     */
    public function __construct(UserClaimMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new UserClaimMvoStateEventId();
        }
    }

    /**
     * @return UserClaimMvoStateEventId
     */
    public function toUserClaimMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getUserClaimIdUserId()
    {
        return $this->value->getUserClaimId()->getUserId();
    }

    /**
     * @param string $userClaimIdUserId
     */
    public function setUserClaimIdUserId($userClaimIdUserId)
    {
        $this->value->getUserClaimId()->setUserId($userClaimIdUserId);
    }

    /**
     * @return integer
     */
    public function getUserClaimIdClaimId()
    {
        return $this->value->getUserClaimId()->getClaimId();
    }

    /**
     * @param integer $userClaimIdClaimId
     */
    public function setUserClaimIdClaimId($userClaimIdClaimId)
    {
        $this->value->getUserClaimId()->setClaimId($userClaimIdClaimId);
    }

    /**
     * @return Long
     */
    public function getUserVersion()
    {
        return $this->value->getUserVersion();
    }

    /**
     * @param Long $userVersion
     */
    public function setUserVersion($userVersion)
    {
        $this->value->setUserVersion($userVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getUserClaimIdUserId(),
            $this->getUserClaimIdClaimId(),
            $this->getUserVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return UserClaimMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setUserClaimIdUserId($pValues[0]);
        $this->setUserClaimIdClaimId($pValues[1]);
        $this->setUserVersion($pValues[2]);
        return $this;
    }

}

